modules/generator/instance.go

```diff
@@ -238,6 +249,7 @@ func (i *instance) updateProcessors() error {
 		}
 	}
 
+	i.histogramsOverride.Store(i.overrides.MetricsGeneratorGenerateNativeHistograms(i.instanceID))
```

Comment:
I like this implementation, it's straightforward and easy to understand.

Only concern is what if the override changes _while_ we are creating the processors. So at the start of `updateProcessors` the override might be `native` and at the end `classic`. This would mean we create the processors with native histograms but then store classic in our flag.

This race condition seems fairly unlikely but I don't think it's impossible.

Reply:
I guess the workaround would be to store the override at the start of `updateProcessors` and then check again at the end, and if it changed restart the whole process? I'm also worried this would make it too convoluted.

Reply:
If such a race happened, this would get corrected on the next reload correct, or are you seeing something more erroneous?

Reply:
Okay thinking about this some more I think we are usually fine. It's only when we go `native` -> `classic` that we need to reload. In the other direction we don't need to reload.

Timeline I was thinking about:

---

Initial situation:
- generate histograms: `native`

\*Some\* overrides change:
- `diffProcessors` detects some of the settings changed -> we start running `updateProcessors`
- `updateProcessors` is running, it's creating processors with `native` histograms

We change the overrides _again_, setting histograms `native` -> `classic`:
- `updateProcessors` is still replacing processors
- the registry picks up the new override and starts creating `classic` histograms now
- `updateProcessors` (finally) finishes and stores `classic` in `histogramsOverride`
  - because the override is currently `classic` we made the assumption that all histograms were created as `classic`, but we don't actually know that 100%

End result:
- some of the histograms will still be created as `native` while the others will already be `classic`
- next time we run `diffProcessors` we don't detect a difference and keep it as-is

---

This is pretty unlikely because it assumes:
- someone changes the overrides twice and both changes need to trigger `updateProcessors` (so `classic`>`native` and then `native`>`classic` would not work)
- the 2nd update doing `native` > `classic` has to happen while `updateProcessors` is busy
  - I think this function is fairly fast, so it's very unlikely

So it's possible to create histograms that are mixed `classic` and `native`, but the window is very small. This would persist until we reload the processors again or restart the instance.

Reply:
To avoid this we would have to pass down whether we want to create classic or native histograms from `instance`, through the processors all the way to the registry. That way we know exactly which type is being created and we don't risk that registry picks up another setting mid-update.
So this requires a lot of changes honestly for a very unlikely scenario.